### PR TITLE
ACL Updates for v2 ResourceAPI

### PR DESCRIPTION
**Pull request by DanStough**
Changes proposed in this PR:
- Adds support for the new ACL Identity Permission. This includes using the new WorkloadIdentity Binding Rule so ACLs have the correct permissions to assume a WorkloadIdentity.

How I've tested this PR: 
* Unit Tests 
* V2 Acceptance Tests are passing in secure mode.

Checklist:
- [X] Tests added
- [ ] ~[CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)~

### CODE REVIEW

File: acceptance/tests/mesh_v2/mesh_inject_test.go [hunk 1]
```diff
@@ -74,9 +79,11 @@ func TestMeshInject_MultiportService(t *testing.T) {
 			require.Len(t, podList.Items, 1)
 			require.Len(t, podList.Items[0].Spec.Containers, 3)
 
-			{
-				// TODO: once ACLs are implemented, only run this block when secure=true and delete the below line and fixture since the default is deny when secure is true.
+			if !secure {
 				k8s.KubectlApplyK(t, ctx.KubectlOptions(t), "../../tests/fixtures/cases/trafficpermissions-deny")
+			}
+
+			{
```
Comment (ndhanushkodi) [inline acceptance/tests/mesh_v2/mesh_inject_test.go:86]: I think there's this extra set of parens around L86-99, maybe we can remove?

File: acceptance/tests/mesh_v2/mesh_inject_test.go [hunk 2]
```diff
@@ -91,6 +98,8 @@ func TestMeshInject_MultiportService(t *testing.T) {
 				})
 			}
 
+			// TODO: consider adding a trafficpermission to a particular port and validate
+
 			// Check connection from static-client to multiport.
 			if cfg.EnableTransparentProxy {
```
Comment (ndhanushkodi) [inline acceptance/tests/mesh_v2/mesh_inject_test.go:104]: Do we want to check at the end of the test deleting the pod and ensuring the ACL token no longer exists in consul? Totally fine to defer that check to a future PR too.

File: control-plane/subcommand/server-acl-init/command_ent_test.go [hunk 3]
```diff
@@ -100,9 +119,15 @@ func TestRun_ConnectInject_SingleDestinationNamespace(t *testing.T) {
 			actRule, _, err := consul.ACL().BindingRuleRead(rules[0].ID, namespaceQuery)
```
Comment (ndhanushkodi) [inline control-plane/subcommand/server-acl-init/command_ent_test.go:119]:
```suggestion
			aclRule, _, err := consul.ACL().BindingRuleRead(rules[0].ID, namespaceQuery)
```
super nit, non blocking

File: control-plane/subcommand/server-acl-init/command_ent_test.go [hunk 4]
```diff
@@ -215,9 +246,15 @@ func TestRun_ConnectInject_NamespaceMirroring(t *testing.T) {
 			actRule, _, err := consul.ACL().BindingRuleRead(rules[0].ID, nil)
```
Comment (ndhanushkodi) [inline control-plane/subcommand/server-acl-init/command_ent_test.go:246]:
```suggestion
			aclRule, _, err := consul.ACL().BindingRuleRead(rules[0].ID, nil)
```
non blocking

File: control-plane/subcommand/server-acl-init/rules.go [hunk 5]
```diff
@@ -351,6 +351,10 @@ partition "{{ .PartitionName }}" {
       policy = "write"
       intentions = "write"
     }
+	identity_prefix "" {
+      policy = "write"
+      intentions = "write"
+    }
 {{- if .EnableNamespaces }}
```
Comment (ndhanushkodi) [inline control-plane/subcommand/server-acl-init/rules.go:358]: I think you might need to change rules_test.go as well

File: acceptance/tests/mesh_v2/mesh_inject_test.go (hunk 2 above)
Comment (DanStough) [reply acceptance/tests/mesh_v2/mesh_inject_test.go]: Yeah, I think I'll defer that to another PR, but I will add a TODO as well.

File: acceptance/tests/mesh_v2/mesh_inject_test.go [hunk 6]
```diff
@@ -74,9 +74,11 @@ func TestMeshInject_MultiportService(t *testing.T) {
 			require.Len(t, podList.Items, 1)
 			require.Len(t, podList.Items[0].Spec.Containers, 3)
 
-			{
-				// TODO: once ACLs are implemented, only run this block when secure=true and delete the below line and fixture since the default is deny when secure is true.
+			if !secure {
 				k8s.KubectlApplyK(t, ctx.KubectlOptions(t), "../../tests/fixtures/cases/trafficpermissions-deny")
+			}
+
+			{
```
Comment (ndhanushkodi) [inline acceptance/tests/mesh_v2/mesh_inject_test.go:81]: Is this set of parens required on L81 and L94? Also non-blocking, I don't think it's worth waiting for another pipeline run to change this

**Review by ndhanushkodi — APPROVED** [review]
Amazing 🎉 and thank you for the flakiness test fixes you made as a part of all this!

File: acceptance/tests/mesh_v2/mesh_inject_test.go (hunk 6 above)
Comment (DanStough) [reply acceptance/tests/mesh_v2/mesh_inject_test.go:81]: Not required. I thought maybe it was scoping this step, but I will remove it.

**Pull request merged**